src/uproot/_dask.py

```diff
@@ -731,142 +743,302 @@ def _get_dask_array_delay_open(
     return dask_dict
 
 
-class _UprootRead:
-    def __init__(
-        self,
-        ttrees,
-        common_keys,
-        common_base_keys,
-        interp_options,
-        form_mapping,
-        rendered_form,
-        original_form=None,
-    ) -> None:
-        self.ttrees = ttrees
-        self.common_keys = common_keys
-        self.common_base_keys = common_base_keys
-        self.interp_options = interp_options
-        self.form_mapping = form_mapping
-        self.rendered_form = rendered_form
-        self.original_form = original_form
+class ImplementsFormMappingInfo(Protocol):
+    @property
+    def behavior(self) -> dict | None:
+        ...
 
-    def __call__(self, i_start_stop):
-        i, start, stop = i_start_stop
+    buffer_key: str | Callable
 
-        if self.form_mapping is not None:
-            awkward = uproot.extras.awkward()
-            dask_awkward = uproot.extras.dask_awkward()
+    def parse_buffer_key(self, buffer_key: str) -> tuple[str, str]:
+        ...
 
-            if set(self.common_keys) != set(self.rendered_form.columns()):
-                actual_form = self.rendered_form.select_columns(self.common_keys)
+    def keys_for_buffer_keys(self, buffer_keys: set[str]) -> set[str]:
+        ...
+
+    def load_buffers(
+        self, tree: HasBranches, keys: set[str], start: int, stop: int, options: Any
+    ) -> Mapping[str, AwkArray]:
+        ...
+
+
+class ImplementsFormMapping(Protocol):
+    def __call__(self, form: Form) -> tuple[Form, ImplementsFormMappingInfo]:
+        ...
+
+
+class TrivialFormMappingInfo(ImplementsFormMappingInfo):
+    def __init__(self, form):
+        awkward = uproot.extras.awkward()
+        assert isinstance(form, awkward.forms.RecordForm)
+
+        self._form = form
+        self._form_key_to_key = self.build_form_key_to_key(form)
+
+    @property
+    def behavior(self) -> None:
+        return None
+
+    @staticmethod
+    def build_form_key_to_key(form: Form) -> dict[str, str | None]:
+        form_key_to_path: dict[str, str | None] = {}
+
+        def impl(form, column_path):
+            # Store columnar path
+            form_key_to_path[form.form_key] = column_path[-1] if column_path else None
+
+            if form.is_union:
+                for _i, entry in enumerate(form.contents):
+                    impl(entry, column_path)
+            elif form.is_indexed:
+                impl(form.content, column_path)
+            elif form.is_list:
+                impl(form.content, column_path)
+            elif form.is_option:
+                impl(form.content, column_path)
+            elif form.is_record:
+                for field in form.fields:
+                    impl(form.content(field), (*column_path, field))
+            elif form.is_unknown or form.is_numpy:
+                pass
             else:
-                actual_form = self.rendered_form
+                raise AssertionError(form)
```

Comment:
This simple recursion builds a mapping form `form_key` to the `key` (depth-0 field) in the TTree